doc/kernel/other/memory_domain.rst

```diff
@@ -0,0 +1,192 @@
+.. _memory_domain:
+
+Memory Domain
+#############
+
+The memory domain APIs are used by unprivileged threads to share data to
+the threads in the same memory and protect sensitive data from threads
+outside their domain. It is not only for improving the security but also
+useful for debugging (unexpected access would cause exception).
+
+.. contents::
+    :local:
+    :depth: 2
+
+Concepts
+********
+
+A memory domain would contain some number of memory partitions.
+A memory partition is a memory region (might be RAM, peripheral registers,
+flash...) with specific attributes (access permission, e.g. privileged
+read/write, unprivileged read-only, execute never...).
+Memory partitions would be defined by set of MPU regions or MMU tables
```

Comment:
How about:

     Memory partitions are defined by a set of underlying MPU regions
     or MMU tables. A thread belongs to a single memory domain at
     any point in time but a memory domain may contain multiple threads.
     Threads in the same memory domain have the same access permissions
     to the memory partitions belong to the memory domain.